iam/cloud-client/snippets/create_role.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
+# Copyright 2024 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START iam_create_role]
+from typing import List
+from google.api_core.exceptions import AlreadyExists, FailedPrecondition
+
+from google.cloud.iam_admin_v1 import (
+    IAMClient,
+    Role,
+)
+
+
+def create_role(
+    project_id: str, role_id: str, permissions: List[str], title: str | None = None
+) -> Role:
+    client = IAMClient()
+
+    parent = f"projects/{project_id}"
+
+    role = Role()
```

Comment:
Either include this role object in the request to create a role, or just not declare it here. No need to make it here to just overwrite it later as it is right now.